aws/container-linux/kubernetes/bastion.tf

```diff
@@ -0,0 +1,166 @@
+resource "aws_autoscaling_group" "bastion" {
+  name = "${var.cluster_name}-bastion ${aws_launch_configuration.bastion.name}"
+
+  # count
+  desired_capacity          = "${var.bastion_count}"
+  min_size                  = "${var.bastion_count}"
+  max_size                  = "${var.bastion_count}"
+  default_cooldown          = 30
+  health_check_grace_period = 30
+
+  # network
+  vpc_zone_identifier = ["${aws_subnet.private.*.id}"]
+
+  # template
+  launch_configuration = "${aws_launch_configuration.bastion.name}"
+
+  # target groups to which instances should be added
+  target_group_arns = [
+    "${aws_lb_target_group.bastion.id}"
+  ]
+
+  lifecycle {
+    # override the default destroy and replace update behavior
+    create_before_destroy = true
+    ignore_changes        = ["image_id"]
+  }
+
+  tags = [{
+    key                 = "Name"
+    value               = "${var.cluster_name}-bastion"
+    propagate_at_launch = true
+  }]
+}
+
+resource "aws_launch_configuration" "bastion" {
+  image_id      = "${data.aws_ami.coreos.image_id}"
```

Comment:
That means using different usernames and an entirely different configuration strategy than the other machines.

Reply:
I do.
We would want different auditing tools, and different dependencies installed on bastion host. 
I understand the extra scope here, so willing to push back to different PR, but since purpose of this instance is vastly different, and is publically accessible over ssh (via NLB) then we should specialize the instance to that purpose